File: java/src/test/java/fyi/oxide/pdf/MarkdownConverterTest.java
/*
 * Copyright 2025-2026 Yury Fedoseev and pdf_oxide contributors.
 * Licensed under MIT OR Apache-2.0.
 */
package fyi.oxide.pdf;

import static org.assertj.core.api.Assertions.assertThat;

import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

class MarkdownConverterTest {

    private static Path fixturesDir;

    @BeforeAll
    static void resolveFixtures() {
        fixturesDir = Paths.get("..")
                .resolve("tests")
                .resolve("fixtures")
                .toAbsolutePath()
                .normalize();
        org.junit.jupiter.api.Assumptions.assumeTrue(
                Files.isDirectory(fixturesDir), "fixtures dir not present: " + fixturesDir);
    }

    @Test
    void toMarkdownProducesHeading() {
        Path hello = fixturesDir.resolve("hello_structure.pdf");
        org.junit.jupiter.api.Assumptions.assumeTrue(Files.exists(hello), "hello_structure.pdf not present");
        try (PdfDocument doc = PdfDocument.open(hello)) {
            String md = MarkdownConverter.toMarkdown(doc, 0);
            assertThat(md).contains("# "); // tagged heading
            assertThat(md).containsIgnoringCase("hello");
        }
    }

    @Test
    void toHtmlProducesContent() {
        Path hello = fixturesDir.resolve("hello_structure.pdf");
        org.junit.jupiter.api.Assumptions.assumeTrue(Files.exists(hello), "hello_structure.pdf not present");
        try (PdfDocument doc = PdfDocument.open(hello)) {
            String html = MarkdownConverter.toHtml(doc, 0);
            assertThat(html).isNotEmpty();
        }
    }

    @Test
    void docConvenienceMethodsMatchConverterStatics() {
        Path hello = fixturesDir.resolve("hello_structure.pdf");
        org.junit.jupiter.api.Assumptions.assumeTrue(Files.exists(hello), "hello_structure.pdf not present");
        try (PdfDocument doc = PdfDocument.open(hello)) {
            assertThat(doc.toMarkdown(0)).isEqualTo(MarkdownConverter.toMarkdown(doc, 0));
            assertThat(doc.toHtml(0)).isEqualTo(MarkdownConverter.toHtml(doc, 0));
            assertThat(doc.toMarkdown()).isEqualTo(MarkdownConverter.toMarkdown(doc));
            assertThat(doc.toHtml()).isEqualTo(MarkdownConverter.toHtml(doc));
        }
    }
}

File: php/tests/Integration/MarkdownConverterTest.php
<?php

/*
 * Copyright 2025-2026 Yury Fedoseev and pdf_oxide contributors.
 * Licensed under MIT OR Apache-2.0.
 */

declare(strict_types=1);

namespace PdfOxide\Tests\Integration;

use PdfOxide\MarkdownConverter;
use PdfOxide\PdfDocument;

/**
 * Smoke tests for {@see MarkdownConverter}, mirroring
 * `java/src/test/java/fyi/oxide/pdf/MarkdownConverterTest.java`.
 */
final class MarkdownConverterTest extends IntegrationTestCase
{
    public function testToMarkdownReturnsString(): void
    {
        $doc = PdfDocument::open($this->fixture('simple.pdf'));
        try {
            $md = MarkdownConverter::toMarkdown($doc, 0);
            $this->assertIsString($md);
        } finally {
            $doc->close();
        }
    }

    public function testToMarkdownAllReturnsString(): void
    {
        $doc = PdfDocument::open($this->fixture('simple.pdf'));
        try {
            $md = MarkdownConverter::toMarkdownAll($doc);
            $this->assertIsString($md);
        } finally {
            $doc->close();
        }
    }

    public function testToHtmlReturnsString(): void
    {
        $doc = PdfDocument::open($this->fixture('simple.pdf'));
        try {
            $html = MarkdownConverter::toHtml($doc, 0);
            $this->assertIsString($html);
        } finally {
            $doc->close();
        }
    }

    public function testDocConvenienceMatchesStaticConverter(): void
    {
        $doc = PdfDocument::open($this->fixture('simple.pdf'));
        try {
            $this->assertSame(MarkdownConverter::toMarkdown($doc, 0), $doc->toMarkdown(0));
            $this->assertSame(MarkdownConverter::toHtml($doc, 0), $doc->toHtml(0));
        } finally {
            $doc->close();
        }
    }
}
